src/Symfony/Component/Translation/Translator.php

```
@@ -171,7 +171,7 @@ public function setLocale($locale)
      */
     public function getLocale()
     {
-        return $this->locale;
+        return $this->locale ?: \Locale::getDefault();
```

Borrowed from https://github.com/symfony/contracts/blob/master/Translation/TranslatorTrait.php#L38 which implies that an empty locale is not a valid locale.

---

if we want to take this route, i would preserve the empty string (as given by the user)

and validate as such in assertValidLocale, rather than silently ignoring it.

I think it's worth a discussion on master.

---

I think we could do this:
For now we deprecate the possibility to set en empty locale in 4.4 and then we add a proper validation in `assertValidLocale` on master.

WDYT?

---

we cannot add new deprecations in a patch release neither :)

---

ah true 🙈